File: HeartBeatSystemDemo/HeartBeatSystemDemo/src/models/Tranquilizer.java
package models;

import iface.DragType;
import iface.Drags;
import thread.DragThread;

public class Tranquilizer implements Drags {
    private static final DragType type = DragType.TRANQUILIZER;
    private int duration;
    public Tranquilizer(int duration){
        this.duration = duration;
    }
    @Override
    public int getDuration() {
        return duration;
    }

    @Override
    public DragType getType() {
        return type;
    }
}

File: UnityProjet/scripts/HeartBeatSystem/models/Tranquilizer.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Tranquilizer : IDrags
{
    public const DragTypes Type = DragTypes.TRANQUILIZER;
    private readonly int Duration;
    public Tranquilizer(int duration)
    {
        this.Duration = duration;
    }
    public int GetDuration()
    {
        return Duration;
    }

    DragTypes IDrags.GetType()
    {
        return Type;
    }
}
